blockstore.go

```diff
@@ -0,0 +1,104 @@
+package filestore
+
+import (
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/ipfs/go-ipfs-blockstore"
+	mh "github.com/multiformats/go-multihash"
+	"golang.org/x/xerrors"
+)
+
+// WithoutDatastore returns a FileManager/Filestore backed entirely by a
+// blockstore without requiring a datastore. It achieves this by coercing the
+// blockstore into a datastore. The resulting blockstore is suitable for usage
+// with DagBuilderHelper with DagBuilderParams#NoCopy=true.
+func WithoutDatastore(bs blockstore.Blockstore) (blockstore.Blockstore, error) {
+	coercer := &dsCoercer{bs}
+
+	// the FileManager stores positional infos (positional mappings) in a
+	// datastore, which in our case is the blockstore coerced into a datastore.
+	fm := NewFileManager(coercer, "/")
```

Comment:
So AFAIK the reason the FileManager takes a path is to prevent secret extraction.

Reply:
I have to confess that I naively thought this was a key prefix for the datastore entries. But it's a root path, and if this were a general utility, it should at least take the path as an arg. I'll close this issue, and we'll keep this component on the Filecoin side.